test/apps/iodemo/io_demo.cc

```diff
@@ -1880,7 +1888,9 @@ class DemoClient : public P2pDemoCommon {
     }
 
     void active_servers_remove(size_t active_index) {
-        assert(active_index < _active_servers.size());
+        if (active_index >= _active_servers.size()) {
+            return;
+        }
```

Comment:
need to avoid calling active_servers_remove if connection was not established. where is called from?